src/tensors/tensor.jl

```diff
@@ -581,27 +581,19 @@ end
 # Real and imaginary parts
 #---------------------------
 function Base.real(t::AbstractTensorMap)
-    # `isreal` for a `Sector` returns true iff the F and R symbols are real. This guarantees
-    # that the real/imaginary part of a tensor `t` can be obtained by just taking
-    # real/imaginary part of the degeneracy data.
-    if isreal(sectortype(t))
-        return TensorMap(real(t.data), codomain(t), domain(t))
-    else
-        msg = "`real` has not been implemented for `$(typeof(t))`."
-        throw(ArgumentError(msg))
-    end
+    # TODO: should we reformulate the old checks in terms of `sectorscalartype`
+    # For anyonic categories, complex numbers typically come into play only in the
+    # braiding and not in the fusion. Hence, it can make sense to work with real
+    # tensors if no braiding is required.
```

Comment:
I wouldn't mind having `real`/`imag` work even if the fusion data is not real. I understand that it might lead to bugs here and there, but for example for singular values, it is quite natural to allow real numbers (since there are no fusion tensors anyways).